ngraph/core/src/descriptor/tensor.cpp

```diff
@@ -90,6 +91,13 @@ const std::string& ov::descriptor::Tensor::get_name() const {
 NGRAPH_SUPPRESS_DEPRECATED_END
 
 const std::unordered_set<std::string>& ov::descriptor::Tensor::get_names() const {
+    if (m_names.empty()) {
```

Comment:
Is there a race condition? Thread 1 executes line 94 (no mutex is needed for this), while Thread 2 in the middle of insertion something on line 97? Both are executing 'get_names() const' and don't expect any data races

I guess some atomic-like approach (with `while(compare_exchange_strong(...))`) could be used here instead of double check of `m_names.empty`, which also may be a bit faster than mutex